### PR TITLE
feat(INT-921): Telegraf for data collection from AWS ECS 

### PR DESCRIPTION
**Pull request by svinothini24**
**PR description:**
This PR is regarding the replacement of cadvisor with the Telegraf agent to scrape the metrics from AWS ECS service.

Please find below the image available in the AWS ECR for testing purpose(The docker image is yet to be made publicly available).
358321641467.dkr.ecr.us-west-2.amazonaws.com/svinothini-wavefront:telegraf-integration

For example, 
"containerDefinitions": [
        {
            "name": "aws-ecs-ec2-wavefront",
            "image": "**358321641467.dkr.ecr.us-west-2.amazonaws.com/svinothini-wavefront:telegraf-integration**",
            "memory": "150",
        }

### CODE REVIEW

**Comment by svinothini24** [comment]
@srinivas-kandula @yogeshprasad @gangadharaswamy Please review

**Comment by svinothini24** [comment]
WIP for README.md

File: aws-ecs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Telegraf agent configured for Wavefront output intended to be used as it's own deployment to collect metrics
+
+FROM telegraf:1.17.2
```
Comment (gangadharaswamy) [inline aws-ecs/Dockerfile:3]: Let's not depend on a specific version, make it generic. Can you test this?
```
FROM telegraf:latest
```

**Review by gangadharaswamy — CHANGES_REQUESTED** [review]
Test with `telegraf:latest` and make the necessary changes.

File: aws-ecs/Dockerfile (hunk 1 above)
Comment (svinothini24) [reply aws-ecs/Dockerfile]: Changed

File: aws-ecs/telegraf.conf [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  ## Timeout for queries.
+  # timeout = "5s"
+
+
+# Configuration for Wavefront proxy to send metrics to
+[[outputs.wavefront]]
+  host = "$WAVEFRONT_PROXY"
+  port = $WAVEFRONT_PROXY_PORT
+  prefix = "$PREFIX"
+  simple_fields = $SIMPLE_FIELDS
```
Comment (yogeshprasad) [inline aws-ecs/telegraf.conf:81]: can we ignore these fields if we are setting the default values. find more information here https://github.com/influxdata/telegraf/tree/master/plugins/outputs/wavefront about  default values

File: aws-ecs/telegraf-example-task-definition.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    "name": "WAVEFRONT_PROXY_PORT",
+                    "value": "2878"
+                },
+                {
+                    "name": "PREFIX",
+                    "value": "telegraf."
+                },
+                {
+                    "name": "METRIC_SEPARATOR",
```
Comment (yogeshprasad) [inline aws-ecs/telegraf-example-task-definition.json:38]: can we remove this from here if it is not intended to be used by the users?

**Review by yogeshprasad — COMMENTED** [review]
Looks good.
Left few minor comments.

File: aws-ecs/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+` [[inputs.ecs]]`
+
+` [[outputs.wavefront]]`
+
+### Additional Arguments
+
+#### Required: The *ip* of your Wavefront proxy
+
+The proxy should be installed on AWS network and accessible by the Docker host machine.
```
Comment (srinivas-kandula) [inline aws-ecs/README.md:38]: Proxy can be installed any where which should be accessible to Docker host.  It is not necessarily on AWS.

File: aws-ecs/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+The amazon-ecs-agent (though it is a container running on the host) is not present in the metadata/stats endpoints.
+
+Below are the additional properties that are required for Telegraf to export stats to your local Wavefront proxy.
+
+### Prerequisites
+1. An AWS account
+2. A Wavefront proxy installed on AWS.
+
+The docker image comprises a telegraf.conf file which is used to configure the input plugin and the output plugin which are ECS and Wavefront respectively.
```
Comment (srinivas-kandula) [inline aws-ecs/README.md:28]: As the telegraf docker is having this config file we don't need to mention in readme.

File: aws-ecs/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
+
+## Exporting AWS ECS Stats to Wavefront using Telegraf
+
+Amazon ECS, input plugin uses the Amazon ECS metadata and stats v2 or [v3][task-metadata-endpoint-v3] API endpoints to gather stats on running containers in a Task.
+
+The wavefronthq/aws-ecs-wavefront container configured with telegraf must be run in the same Task as the workload it is inspecting to export stats to Wavefront (https://wavefront.com) output plugin.
```
Comment (srinivas-kandula) [inline aws-ecs/README.md:6]: I think we can shorten this and just mention that this task uses telegraf ecs plugin and provide link to plugin.

File: aws-ecs/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+`{`
+
+    "name": "ECS_CONTAINER_METADATA_ENDPOINT",
+    "value": "http://169.254.170.2"
+`}`
+
+The amazon-ecs-agent (though it is a container running on the host) is not present in the metadata/stats endpoints.
+
+Below are the additional properties that are required for Telegraf to export stats to your local Wavefront proxy.
```
Comment (srinivas-kandula) [inline aws-ecs/README.md:22]: Which additional properties and where are they?

**Review by srinivas-kandula — CHANGES_REQUESTED** [review]

File: aws-ecs/README.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+This task makes use of telegraf-ecs plugin and export stats to Wavefront (https://wavefront.com).
+
+The telegraf agent running on ECS instance enables the input plugin and the output plugin which are ECS and Wavefront respectively.
+
+```
+ [[inputs.ecs]]
+ 
+ [[outputs.wavefront]]
+```
+For further reference, please refer to the documentation - https://github.com/influxdata/telegraf/blob/release-1.18/plugins/inputs/ecs/README.md
```
Comment (srinivas-kandula) [inline aws-ecs/README.md:15]: Change line number 15 to:- For further reference, please refer to the telegraf [ecs plugin](https://github.com/influxdata/telegraf/blob/release-1.18/plugins/inputs/ecs/README.md)

File: aws-ecs/README.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
+
+## Exporting AWS ECS Stats to Wavefront using Telegraf
+
+Amazon ECS, input plugin uses the Amazon ECS metadata and stats v2 or [v3][task-metadata-endpoint-v3] API endpoints to gather stats on running containers in a Task.
+
+This task makes use of telegraf-ecs plugin and export stats to Wavefront (https://wavefront.com).
```
Comment (srinivas-kandula) [inline aws-ecs/README.md:6]: Wrap Wavefront in square brackets to get the hyperlink.

**Comment by srinivas-kandula** [comment]
I have added two minor comments rest LGTM.

**Review by srinivas-kandula — APPROVED** [review]

**Review by sushantdewan123 — APPROVED** [review]

**Review by gangadharaswamy — APPROVED** [review]
LGTM

**Review by yogeshprasad — APPROVED** [review]
LGTM

**Pull request merged**